### ISSUE TITLE
apk closes on android

### ISSUE THREAD

**Issue by andrewm41** [kivy/kivy#4462]
Hi, I'm trying to run simple apk on android emulator, here is the main.py code:

``` python
#!/usr/bin/env python
# -*- coding: utf-8 -*-

from datetime import datetime

from kivy.app import App
from kivy.uix.button import Button

class WordsApp(App):
    def build(self):
        return Button(text=u'Push me',
                      on_release=self.print_something)

    def print_something(self, w=None):
        print datetime.now()


if __name__ == '__main__':
    WordsApp().run()
```

most of the times it shows "kivy loading" image and crashes, sometimes I can launch it but I need for it launch another app on emulator close it and start my apk, but this method is about 30-50% chance to start apk

here is the logcat

```
07-12 10:54:33.682  1383  1383 V SDL     : surfaceCreated()
07-12 10:54:33.683  1383  1383 V SDL     : surfaceChanged()
07-12 10:54:33.683  1383  1383 V SDL     : pixel format RGB_565
07-12 10:54:33.688  1383  1383 V SDL     : Window size:396x320
07-12 10:54:33.837  1383  1401 I SDL     : SDL_Android_Init()
07-12 10:54:33.838  1383  1401 I SDL     : SDL_Android_Init() finished!
07-12 10:54:33.848   528   528 I Choreographer: Skipped 44 frames!  The application may be doing too much work on its main thread.
07-12 10:54:33.861  1383  1401 I python  : Initialize Python for Android
07-12 10:54:33.870  1383  1401 I python  : Changing directory to the one provided by ANDROID_ARGUMENT
07-12 10:54:33.870  1383  1401 I python  : /data/user/0/org.minimal.minimal/files
07-12 10:54:33.870  1383  1401 I python  : Preparing to initialize python
07-12 10:54:33.870  1383  1401 I python  : crystax_python does not exist
07-12 10:54:34.033   270   299 V WindowManager: Layouts looping: On entry to LockedInner, mPendingLayoutChanges = 0x1
07-12 10:54:34.033   270   299 V WindowManager: Layouts looping: after finishPostLayoutPolicyLw, mPendingLayoutChanges = 0x0
07-12 10:54:34.034   270   299 V WindowManager: Layouts looping: mLayoutNeeded, mPendingLayoutChanges = 0x1
07-12 10:54:34.037   270   299 V WindowManager: Layouts looping: On entry to LockedInner, mPendingLayoutChanges = 0x1
07-12 10:54:34.038   270   299 V WindowManager: Layouts looping: after finishPostLayoutPolicyLw, mPendingLayoutChanges = 0x0
07-12 10:54:34.038   270   299 V WindowManager: Layouts looping: mLayoutNeeded, mPendingLayoutChanges = 0x1
07-12 10:54:34.052   270   299 E WindowManager: Performed 6 layouts in a row. Skipping
07-12 10:54:34.082  1383  1383 V SDL     : onWindowFocusChanged(): true
07-12 10:54:34.103   270   299 W WindowManager: Window freeze timeout expired.
07-12 10:54:34.103   270   299 W WindowManager: Force clearing orientation change: Window{93353ca u0 Starting org.minimal.minimal}
07-12 10:54:34.104   270   299 W WindowManager: Force clearing orientation change: Window{24474c0 u0 com.android.systemui.ImageWallpaper}
07-12 10:54:34.142   270   299 I ActivityManager: Displayed org.minimal.minimal/org.kivy.android.PythonActivity: +2s459ms
07-12 10:54:34.146   270   299 I WindowManager: Screen frozen for +2s69ms due to Window{93353ca u0 Starting org.minimal.minimal}
07-12 10:54:34.280  1383  1401 I python  : Initialized python
07-12 10:54:34.280  1383  1401 I python  : AND: Init threads
07-12 10:54:34.344  1383  1401 I python  : testing python print redirection
07-12 10:54:34.375  1383  1401 I python  : Setting up python from ANDROID_PRIVATE
07-12 10:54:34.477  1383  1401 I python  : ('Android path', ['/data/user/0/org.minimal.minimal/files/lib/python27.zip', '/data/user/0/org.minimal.minimal/files/lib/python2.7/', '/data/user/0/org.minimal.minimal/files/lib/python2.7/lib-dynload/', '/data/user/0/org.minimal.minimal/files/lib/python2.7/site-packages/', '/data/user/0/org.minimal.minimal/files'])
07-12 10:54:34.787  1383  1401 I python  : ('os.environ is', {'ANDROID_APP_PATH': '/data/user/0/org.minimal.minimal/files', 'EXTERNAL_STORAGE': '/sdcard', 'ANDROID_ENTRYPOINT': 'main.pyo', 'ANDROID_ASSETS': '/system/app', 'PYTHONPATH': '/data/user/0/org.minimal.minimal/files:/data/user/0/org.minimal.minimal/files/lib', 'ANDROID_ARGUMENT': '/data/user/0/org.minimal.minimal/files', 'ANDROID_ROOT': '/system', 'SYSTEMSERVERCLASSPATH': '/system/framework/services.jar:/system/framework/ethernet-service.jar:/system/framework/wifi-service.jar', 'ANDROID_DATA': '/data', 'PYTHONHOME': '/data/user/0/org.minimal.minimal/files', 'ANDROID_SOCKET_zygote': '10', 'BOOTCLASSPATH': '/system/framework/core-libart.jar:/system/framework/conscrypt.jar:/system/framework/okhttp.jar:/system/framework/core-junit.jar:/system/framework/bouncycastle.jar:/system/framework/ext.jar:/system/framework/framework.jar:/system/framework/telephony-common.jar:/system/framework/voip-common.jar:/system/framework/ims-common.jar:/system/framework/apache-xml.jar:/system/framework/org.apache.http.legacy.boot.jar', 'ANDROID_PROPERTY_WORKSPACE': '9,0', 'ASEC_MOUNTPOINT': '/mnt/asec', 'PATH': '/sbin:/vendor/bin:/system/sbin:/system/bin:/system/xbin', 'ANDROID_STORAGE': '/storage', 'ANDROID_BOOTLOGO': '1', 'ANDROID_PRIVATE': '/data/user/0/org.minimal.minimal/files', 'PYTHON_NAME': 'python'})
07-12 10:54:34.790  1383  1401 I python  : ('Android kivy bootstrap done. __name__ is', '__main__')
07-12 10:54:35.018  1383  1401 I python  : ['/data/user/0/org.minimal.minimal/files/lib/python2.7/site-packages', '/data/user/0/org.minimal.minimal/files/lib/site-python']
07-12 10:54:35.018  1383  1401 I python  : AND: Ran string
07-12 10:54:35.018  1383  1401 I python  : Run user program, change dir and execute entrypoint
07-12 10:54:35.019  1383  1401 I python  : main.py
07-12 10:54:35.635  1383  1401 I python  : [INFO   ] [Logger      ] Record log in /data/user/0/org.minimal.minimal/files/.kivy/logs/kivy_16-07-12_1.txt
07-12 10:54:35.636  1383  1401 I python  : [INFO   ] [Kivy        ] v1.9.2-dev0
07-12 10:54:35.637  1383  1401 I python  : [INFO   ] [Python      ] v2.7.2 (default, Jul 11 2016, 22:43:23) 
07-12 10:54:35.637  1383  1401 I python  : [GCC 4.8]
07-12 10:54:35.677  1383  1401 W linker  : /data/data/org.minimal.minimal/files/lib/python2.7/site-packages/kivy/_clock.so: is missing DT_SONAME will use basename as a replacement: "_clock.so"
07-12 10:54:35.726  1383  1401 W linker  : /data/data/org.minimal.minimal/files/lib/python2.7/site-packages/kivy/_event.so: is missing DT_SONAME will use basename as a replacement: "_event.so"
07-12 10:54:35.737  1383  1401 W linker  : /data/data/org.minimal.minimal/files/lib/python2.7/site-packages/kivy/properties.so: is missing DT_SONAME will use basename as a replacement: "properties.so"
07-12 10:54:35.778  1383  1401 I python  : [INFO   ] [Factory     ] 193 symbols loaded
07-12 10:54:36.176   270   299 I art     : Starting a blocking GC Explicit
07-12 10:54:36.339   270   299 I art     : Explicit concurrent mark sweep GC freed 7543(492KB) AllocSpace objects, 1(20KB) LOS objects, 33% free, 5MB/7MB, paused 4.592ms total 159.470ms
07-12 10:54:37.200  1010  1010 I Choreographer: Skipped 93 frames!  The application may be doing too much work on its main thread.
07-12 10:54:37.208   270   278 W art     : Long monitor contention event with owner method=long android.content.res.AssetManager.openXmlAssetNative(int, java.lang.String) from AssetManager.java:4294967294 waiters=0 for 839ms
07-12 10:54:38.360   270   288 W VoiceInteractionManagerService: no available voice recognition services found for user 0
07-12 10:54:38.420  1383  1401 W linker  : /data/data/org.minimal.minimal/files/lib/python2.7/site-packages/kivy/graphics/instructions.so: is missing DT_SONAME will use basename as a replacement: "instructions.so"
07-12 10:54:38.429  1383  1401 W linker  : /data/data/org.minimal.minimal/files/lib/python2.7/site-packages/kivy/graphics/buffer.so: is missing DT_SONAME will use basename as a replacement: "buffer.so"
07-12 10:54:38.441  1383  1401 W linker  : /data/data/org.minimal.minimal/files/lib/python2.7/site-packages/kivy/graphics/vertex.so: is missing DT_SONAME will use basename as a replacement: "vertex.so"
07-12 10:54:38.454  1383  1401 W linker  : /data/data/org.minimal.minimal/files/lib/python2.7/site-packages/kivy/graphics/vbo.so: is missing DT_SONAME will use basename as a replacement: "vbo.so"
07-12 10:54:38.467  1383  1401 W linker  : /data/data/org.minimal.minimal/files/lib/python2.7/site-packages/kivy/graphics/compiler.so: is missing DT_SONAME will use basename as a replacement: "compiler.so"
07-12 10:54:38.478  1383  1401 W linker  : /data/data/org.minimal.minimal/files/lib/python2.7/site-packages/kivy/graphics/transformation.so: is missing DT_SONAME will use basename as a replacement: "transformation.so"
07-12 10:54:38.486  1383  1401 W linker  : /data/data/org.minimal.minimal/files/lib/python2.7/site-packages/kivy/graphics/shader.so: is missing DT_SONAME will use basename as a replacement: "shader.so"
07-12 10:54:38.498  1383  1401 W linker  : /data/data/org.minimal.minimal/files/lib/python2.7/site-packages/kivy/graphics/texture.so: is missing DT_SONAME will use basename as a replacement: "texture.so"
07-12 10:54:38.512  1383  1401 W linker  : /data/data/org.minimal.minimal/files/lib/python2.7/site-packages/kivy/graphics/context_instructions.so: is missing DT_SONAME will use basename as a replacement: "context_instructions.so"
07-12 10:54:38.692  1383  1401 W linker  : /data/data/org.minimal.minimal/files/lib/python2.7/site-packages/kivy/core/image/_img_sdl2.so: is missing DT_SONAME will use basename as a replacement: "_img_sdl2.so"
07-12 10:54:38.715  1383  1401 I python  : [INFO   ] [Image       ] Providers: img_tex, img_dds, img_gif, img_sdl2 (img_pil, img_ffpyplayer ignored)
07-12 10:54:38.721  1383  1401 W linker  : /data/data/org.minimal.minimal/files/lib/python2.7/site-packages/kivy/graphics/fbo.so: is missing DT_SONAME will use basename as a replacement: "fbo.so"
07-12 10:54:38.734  1383  1401 W linker  : /data/data/org.minimal.minimal/files/lib/python2.7/site-packages/kivy/graphics/context.so: is missing DT_SONAME will use basename as a replacement: "context.so"
07-12 10:54:38.748  1383  1401 W linker  : /data/data/org.minimal.minimal/files/lib/python2.7/site-packages/kivy/graphics/opengl.so: is missing DT_SONAME will use basename as a replacement: "opengl.so"
07-12 10:54:38.800  1383  1401 W linker  : /data/data/org.minimal.minimal/files/lib/python2.7/site-packages/kivy/graphics/opengl_utils.so: is missing DT_SONAME will use basename as a replacement: "opengl_utils.so"
07-12 10:54:38.824  1383  1401 W linker  : /data/data/org.minimal.minimal/files/lib/python2.7/site-packages/kivy/graphics/vertex_instructions.so: is missing DT_SONAME will use basename as a replacement: "vertex_instructions.so"
07-12 10:54:38.841  1383  1401 W linker  : /data/data/org.minimal.minimal/files/lib/python2.7/site-packages/kivy/graphics/stencil_instructions.so: is missing DT_SONAME will use basename as a replacement: "stencil_instructions.so"
07-12 10:54:38.858  1383  1401 W linker  : /data/data/org.minimal.minimal/files/lib/python2.7/site-packages/kivy/graphics/gl_instructions.so: is missing DT_SONAME will use basename as a replacement: "gl_instructions.so"
07-12 10:54:38.872  1383  1401 W linker  : /data/data/org.minimal.minimal/files/lib/python2.7/site-packages/kivy/graphics/scissor_instructions.so: is missing DT_SONAME will use basename as a replacement: "scissor_instructions.so"
07-12 10:54:38.897  1383  1401 W linker  : /data/data/org.minimal.minimal/files/lib/python2.7/site-packages/kivy/weakproxy.so: is missing DT_SONAME will use basename as a replacement: "weakproxy.so"
07-12 10:54:38.921  1383  1401 W linker  : /data/data/org.minimal.minimal/files/lib/python2.7/site-packages/kivy/core/text/text_layout.so: is missing DT_SONAME will use basename as a replacement: "text_layout.so"
07-12 10:54:38.936  1383  1401 W linker  : /data/data/org.minimal.minimal/files/lib/python2.7/site-packages/kivy/core/text/_text_sdl2.so: is missing DT_SONAME will use basename as a replacement: "_text_sdl2.so"
07-12 10:54:38.945  1383  1401 I python  : [INFO   ] [Text        ] Provider: sdl2
07-12 10:54:39.002  1383  1401 W linker  : /data/data/org.minimal.minimal/files/lib/python2.7/site-packages/jnius/jnius.so: is missing DT_SONAME will use basename as a replacement: "jnius.so"
07-12 10:54:42.200  1383  1401 W linker  : /data/data/org.minimal.minimal/files/lib/python2.7/site-packages/kivy/core/window/_window_sdl2.so: is missing DT_SONAME will use basename as a replacement: "_window_sdl2.so"
07-12 10:54:42.295  1383  1401 I python  : [INFO   ] [OSC         ] using <thread> for socket
07-12 10:54:42.309  1383  1401 I python  : [ERROR  ] [Input       ] AndroidJoystick is not supported by your version of linux
07-12 10:54:42.311  1383  1401 I python  : Traceback (most recent call last):
07-12 10:54:42.312  1383  1401 I python  :   File "/data/user/0/org.minimal.minimal/files/lib/python2.7/site-packages/kivy/input/providers/__init__.py", line 57, in <module>
07-12 10:54:42.312  1383  1401 I python  :     import kivy.input.providers.androidjoystick
07-12 10:54:42.312  1383  1401 I python  :   File "/data/user/0/org.minimal.minimal/files/lib/python2.7/site-packages/kivy/input/providers/androidjoystick.py", line 20, in <module>
07-12 10:54:42.312  1383  1401 I python  :     raise Exception('android lib not found.')
07-12 10:54:42.312  1383  1401 I python  : Exception: android lib not found.
07-12 10:54:42.318  1383  1401 I python  : [INFO   ] [Window      ] Provider: sdl2
07-12 10:54:42.414  1383  1401 D         : HostConnection::get() New Host Connection established 0xa3894d70, tid 1401
07-12 10:54:42.480  1010  1010 I art     : Waiting for a blocking GC DisableMovingGc
07-12 10:54:42.494  1383  1401 I python  : [INFO   ] [GL          ] OpenGL version <OpenGL ES 2.0 (4.5.0 NVIDIA 361.28)>
07-12 10:54:42.496  1383  1401 I python  : [INFO   ] [GL          ] OpenGL vendor <Google (NVIDIA Corporation)>
07-12 10:54:42.498  1383  1401 I python  : [INFO   ] [GL          ] OpenGL renderer <Android Emulator OpenGL ES Translator (GeForce GTX 960/PCIe/SSE2)>
07-12 10:54:42.504  1383  1401 I python  : [INFO   ] [GL          ] OpenGL parsed version: 2, 0
07-12 10:54:42.507  1383  1401 I python  : [INFO   ] [GL          ] Texture max size <16384>
07-12 10:54:42.509  1383  1401 I python  : [INFO   ] [GL          ] Texture max units <32>
07-12 10:54:42.556  1010  1010 I art     : WaitForGcToComplete blocked for 75.504ms for cause DisableMovingGc
07-12 10:54:42.557  1010  1010 I art     : Starting a blocking GC DisableMovingGc
07-12 10:54:42.562  1383  1401 I python  :  Traceback (most recent call last):
07-12 10:54:42.563  1383  1401 I python  :    File "/home/andrew/local/buildozer/minimal/.buildozer/android/platform/build/build/other_builds/python2-sqlite3/armeabi-v7a/python2/python-install/lib/python2.7/logging/__init__.py", line 865, in emit
07-12 10:54:42.586  1383  1401 I python  :      stream.write(fs % msg.encode("UTF-8"))
07-12 10:54:42.588  1383  1401 I python  :  UnicodeDecodeError: 'ascii' codec can't decode byte 0xa0 in position 43: ordinal not in range(128)
07-12 10:54:42.590  1383  1401 I python  :  Logged from file __init__.py, line 941
07-12 10:54:42.627  1383  1401 I python  : [CRITICAL] [Window      ] Unable to find any valuable Window provider.
07-12 10:54:42.627  1383  1401 I python  : sdl2 - UnicodeDecodeError: 'utf8' codec can't decode byte 0x90 in position 0: invalid start byte
07-12 10:54:42.627  1383  1401 I python  :   File "/data/user/0/org.minimal.minimal/files/lib/python2.7/site-packages/kivy/core/__init__.py", line 67, in core_select_lib
07-12 10:54:42.627  1383  1401 I python  :     cls = cls()
07-12 10:54:42.627  1383  1401 I python  :   File "/data/user/0/org.minimal.minimal/files/lib/python2.7/site-packages/kivy/core/window/window_sdl2.py", line 139, in __init__
07-12 10:54:42.627  1383  1401 I python  :     super(WindowSDL, self).__init__()
07-12 10:54:42.627  1383  1401 I python  :   File "/data/user/0/org.minimal.minimal/files/lib/python2.7/site-packages/kivy/core/window/__init__.py", line 789, in __init__
07-12 10:54:42.627  1383  1401 I python  :     self.create_window()
07-12 10:54:42.627  1383  1401 I python  :   File "/data/user/0/org.minimal.minimal/files/lib/python2.7/site-packages/kivy/core/window/window_sdl2.py", line 283, in create_window
07-12 10:54:42.627  1383  1401 I python  :     super(WindowSDL, self).create_window()
07-12 10:54:42.627  1383  1401 I python  :   File "/data/user/0/org.minimal.minimal/files/lib/python2.7/site-packages/kivy/core/window/__init__.py", line 941, in create_window
07-12 10:54:42.627  1383  1401 I python  :     self.render_context = RenderContext()
07-12 10:54:42.627  1383  1401 I python  :   File "kivy/graphics/instructions.pyx", line 754, in kivy.graphics.instructions.RenderContext.__init__ (kivy/graphics/instructions.c:10606)
07-12 10:54:42.627  1383  1401 I python  :   File "kivy/graphics/shader.pyx", line 186, in kivy.graphics.shader.Shader.__init__ (kivy/graphics/shader.c:3836)
07-12 10:54:42.627  1383  1401 I python  :   File "kivy/graphics/shader.pyx", line 695, in kivy.graphics.shader.Shader.vs.__set__ (kivy/graphics/shader.c:10943)
07-12 10:54:42.627  1383  1401 I python  :   File "kivy/graphics/shader.pyx", line 557, in kivy.graphics.shader.Shader.build_vertex (kivy/graphics/shader.c:8826)
07-12 10:54:42.628  1383  1401 I python  :   File "kivy/graphics/shader.pyx", line 580, in kivy.graphics.shader.Shader.link_program (kivy/graphics/shader.c:9141)
07-12 10:54:42.628  1383  1401 I python  :   File "kivy/graphics/shader.pyx", line 634, in kivy.graphics.shader.Shader.get_program_log (kivy/graphics/shader.c:9788)
07-12 10:54:42.628  1383  1401 I python  :   File "/home/andrew/local/buildozer/minimal/.buildozer/android/platform/build/build/other_builds/python2-sqlite3/armeabi-v7a/python2/python-install/lib/python2.7/encodings/utf_8.py", line 16, in decode
07-12 10:54:42.628  1383  1401 I python  :     return codecs.utf_8_decode(input, errors, True)
07-12 10:54:42.630  1383  1401 I python  : [CRITICAL] [App         ] Unable to get a Window, abort.
07-12 10:54:42.843   270  1024 D GraphicsStats: Buffer count: 2
07-12 10:54:42.844   270  1024 I WindowState: WIN DEATH: Window{db871e9 u0 org.minimal.minimal/org.kivy.android.PythonActivity}
07-12 10:54:42.844   270  1024 W WindowManager: Force-removing child win Window{f38d5cc u0 SurfaceView} from container Window{db871e9 u0 org.minimal.minimal/org.kivy.android.PythonActivity}
07-12 10:54:42.883   270   459 W WindowManager: Failed looking up window
07-12 10:54:42.883   270   459 W WindowManager: java.lang.IllegalArgumentException: Requested window android.os.BinderProxy@d73ae6e does not exist
07-12 10:54:42.883   270   459 W WindowManager:         at com.android.server.wm.WindowManagerService.windowForClientLocked(WindowManagerService.java:8723)
07-12 10:54:42.883   270   459 W WindowManager:         at com.android.server.wm.WindowManagerService.windowForClientLocked(WindowManagerService.java:8714)
07-12 10:54:42.883   270   459 W WindowManager:         at com.android.server.wm.WindowState$DeathRecipient.binderDied(WindowState.java:1209)
07-12 10:54:42.883   270   459 W WindowManager:         at android.os.BinderProxy.sendDeathNotice(Binder.java:558)
07-12 10:54:42.884   270   459 I WindowState: WIN DEATH: null
07-12 10:54:42.889    74    74 I Zygote  : Process 1383 exited cleanly (1)
07-12 10:54:42.891   270   521 I ActivityManager: Process org.minimal.minimal (pid 1383) has died
07-12 10:54:42.892   270   521 W ActivityManager: Force removing ActivityRecord{4e84140 u0 org.minimal.minimal/org.kivy.android.PythonActivity t6}: app died, no saved state
07-12 10:54:42.949    60    60 E EGL_emulation: tid 60: eglCreateSyncKHR(1243): error 0x3004 (EGL_BAD_ATTRIBUTE)
07-12 10:54:42.970   270   521 I ActivityManager: Config changes=480 {1.0 310mcc260mnc en_US ldltr sw160dp w160dp h168dp 320dpi smll port finger qwerty/v/v -nav/h s.11}
```

say if it is helpfull to post bigger log slice

**Comment by GadgeurX** [comment]
Same issue, have you more information ?

**Comment by andrewm41** [comment]
It was decode error for me.

I use buildozer to deploy app.

It is fixed in Kivy 1.9.1 but buildozer downloaded Kivy 1.9.0 when building app.

You can check this by printing Kivy version on app start. Buildozer doesn't use your local version of Kivy but download its own copy.

I fixed this with replacing .buildozer/android/platform/build/dists/minimal/private/lib/python2.7/site-packages/kivy/graphics/shader.so with 1.9.1 version

here is what I did, note that "minimal" is the name of my app, you should change it accordingly (consider to be in the folder with your app), and make your app at least once to let buildozer to download files:

wget -O .buildozer/android/platform/build/dists/minimal/private/lib/python2.7/site-packages/kivy/graphics/shader.pyx https://raw.githubusercontent.com/kivy/kivy/master/kivy/graphics/shader.pyx

create file .buildozer/android/platform/build/dists/minimal/private/lib/python2.7/site-packages/build_shader.py
with content (file name is optional)

from distutils.core import setup
from Cython.Build import cythonize
setup(
    ext_modules = cythonize("kivy/graphics/shader.pyx")
)

and execute:
cd .buildozer/android/platform/build/dists/minimal/private/lib/python2.7/site-packages/ && python build_shader.py build_ext --inplace && cd -

and the next time you deploy your app to device, it will contain updated cython compiled file
